hermeto/core/models/input.py

```diff
@@ -120,6 +122,60 @@ def _validate_ssl_options(self) -> Self:
         return self
 
 
+class BinaryFilter(pydantic.BaseModel, extra="forbid"):
+    """Represents a filter for binary packages.
+
+    An empty filters set means "accept all" (no filtering).
+    A populated filters set means "accept only these specific items".
+    """
+
+    filters: set[str] = set()
+
+    @property
+    def is_all(self) -> bool:
+        """Return True if this filter accepts all (no filtering)."""
+        return not self.filters
+
+    def __str__(self) -> str:
+        return ":all:" if self.is_all else ",".join(sorted(self.filters))
+
+    def __repr__(self) -> str:
+        return (
+            f"<{type(self).__name__} "
+            f"filters={':all:' if self.is_all else ','.join(sorted(self.filters))}>"
+        )
+
+
+def _parse_binary_filter(value: Any) -> BinaryFilter:
```

Comment:
This feels like a classmethod for BinaryFilter, is there any reason to keep this separate from the class?